File: lg5-spring-kafka/lg5-spring-kafka-consumer/src/main/java/com/lg5/spring/kafka/consumer/KafkaConsumer.java
package com.lg5.spring.kafka.consumer;

import org.apache.avro.specific.SpecificRecordBase;

import java.util.List;

public interface KafkaConsumer<T extends SpecificRecordBase> {
    void receive(final List<T> messages, final List<String> keys, final List<Integer> partitions, final List<Long> offsets);
}

File: lg5-spring-kafka/lg5-spring-kafka-consumer/src/main/kotlin/com/lg5/spring/kafka/consumer/KafkaConsumer.kt
package com.lg5.spring.kafka.consumer

import org.apache.avro.specific.SpecificRecordBase


interface KafkaConsumer {
    fun receive(
        messages: List<SpecificRecordBase>,
        keys: List<String?>?,
        partitions: List<Int?>?,
        offsets: List<Long?>?
    )
}